janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/CustomAnalyzer.java

```diff
@@ -0,0 +1,121 @@
+        this.informations = informations;
+    }
+
+    @Override
+    protected Analyzer getWrappedAnalyzer(String fieldName) {
+        Mapping mapping = null;
+        String stringAnalyzerString = null;
+        String textAnalyzerString = null;
+
+
+        Parameter[] parameters = null;
+        Class<?> clazz = null;
+
+
+        KeyInformation keyInformation = informations.get(store, fieldName);
+        if (keyInformation != null) {
+            parameters = keyInformation.getParameters();
+            clazz = keyInformation.getDataType();
+        }
+
+        if (parameters != null) {
+            // if mapping isn't present in parameters, we use Mapping.DEFAULT
+            mapping = ParameterType.MAPPING.findParameter(parameters, Mapping.DEFAULT);
+            stringAnalyzerString = ParameterType.STRING_ANALYZER.findParameter(parameters, null);
+            textAnalyzerString = ParameterType.TEXT_ANALYZER.findParameter(parameters, null);
+        }
+      
+        String analyzerString = null;
+
+        if (clazz != null && String.class.isAssignableFrom(clazz) && mapping != null) {
+            // at the moment, we only try to support custom analyzers for string data.
+            // everything else falls through a StandardAnalyzer as was the case before
+            switch (mapping) {
+                case TEXTSTRING:
+                    throw new RuntimeException("TextString is an unsupported mapping for string data & custom analyzers");
+                case PREFIX_TREE:
+                    throw new RuntimeException("Prefix-tree is an unsupported mapping for string data & custom analyzers");
+                case STRING:
+                    if (stringAnalyzerString != null) analyzerString = stringAnalyzerString; else analyzerString = org.apache.lucene.analysis.core.KeywordAnalyzer.class.getName();
+                    break;
+                case TEXT:
+                case DEFAULT:// TEXT
+                    analyzerString = textAnalyzerString;
+                    break;
+            }
+        }
+
+        // the default (& fall through) Analyzer is a StandardAnalyzer
+        if ((clazz == null || !String.class.isAssignableFrom(clazz)) || mapping == null || analyzerString == null) {
+            analyzerString = org.apache.lucene.analysis.standard.StandardAnalyzer.class.getName();
+        }
+
+        final String analyzerStringFinal = analyzerString;
+
+        Analyzer analyzer = analyzers.computeIfAbsent(analyzerString, t -> {
+            try {
+                Class classDefinition = Class.forName(analyzerStringFinal);
+                return (Analyzer) classDefinition.newInstance();
+            } catch (Exception e) {
+                throw new RuntimeException("Analyzer cannot be instanciated for class " + analyzerStringFinal, e);
+            }
+        });
+        if (analyzer == null) {
+            throw new RuntimeException("Analyzer cannot be instanciated for class " + analyzerString);
```

Comment:
Instead of Runtime, can you use PermanentBackendException, because it's a problem of configuration